src/components/app-configure/auth/SocialAuth.vue

```diff
@@ -0,0 +1,172 @@
+<script lang="ts" setup>
+import { computed, type ComputedRef } from 'vue'
+import { useStore } from 'vuex'
+
+import CopyIcon from '@/assets/iconography/copy.svg'
+import LinkIcon from '@/assets/iconography/link.svg'
+import SettingCard from '@/components/app-configure/SettingCard.vue'
+import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import {
+  socialLogins,
+  type SocialAuth,
+  type SocialAuthVerifier,
+  type SocialAuthField,
+} from '@/utils/constants'
+import copyToClipboard from '@/utils/copyToClipboard'
+
+const store = useStore()
+
+const socialAuthTypes: SocialAuthField[] = [
+  'clientId',
+  'clientSecret',
+  'redirectUri',
+]
+
+const authDetails: ComputedRef<SocialAuth[]> = computed(
+  () => store.getters.socialAuth
+)
+
+function getAuthIndex(
+  authDetailsList: SocialAuth[],
+  verifier: SocialAuthVerifier
+): number {
+  return authDetailsList.findIndex(
+    (authDetail: SocialAuth) => authDetail.verifier === verifier
+  )
+}
+
+function getClient(verifier: SocialAuthVerifier): SocialAuth | undefined {
+  const tempAuthDetails: SocialAuth[] = [...authDetails.value]
+  let existingAuthIndex = getAuthIndex(tempAuthDetails, verifier)
+  if (existingAuthIndex !== -1) {
+    return tempAuthDetails[existingAuthIndex]
+  }
+}
+
+function handleUpdate(
+  verifier: SocialAuthVerifier,
+  inputType: SocialAuthField,
+  value: string
+) {
+  let tempAuthDetails: SocialAuth[] = [...authDetails.value]
+  const existingAuthIndex = getAuthIndex(tempAuthDetails, verifier)
+  if (existingAuthIndex !== -1) {
+    const existingAuth = { ...tempAuthDetails[existingAuthIndex] }
+    const otherInputTypes: SocialAuthField[] = socialAuthTypes.filter(
+      (el) => el !== inputType
+    )
+    if (
+      !value &&
+      !existingAuth[otherInputTypes[0]] &&
+      !existingAuth[otherInputTypes[1]]
+    ) {
+      tempAuthDetails = tempAuthDetails.filter(
+        (authDetail) => authDetail.verifier !== existingAuth.verifier
+      )
+    } else {
+      existingAuth[inputType] = value
+      tempAuthDetails[existingAuthIndex] = existingAuth
```

Comment:
@ajithranka @karthik-durai I needed to do this as I was getting a vuex warning when using `tempAuthDetails[existingAuthIndex][inputType] = value`

```
[Vue warn]: Unhandled error during execution of watcher callback
```

When I wrapped the code inside the function in try catch block, I got more detailed error
```
Error: [vuex] do not mutate vuex store state outside mutation handlers.
    at assert (vuex.esm-bundler.js:83:27)
    at watch.deep (vuex.esm-bundler.js:347:7)
    at callWithErrorHandling (runtime-core.esm-bundler.js:155:22)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js:164:21)
    at ReactiveEffect.job [as scheduler] (runtime-core.esm-bundler.js:1779:17)
    at triggerEffects (reactivity.esm-bundler.js:382:24)
    at trigger (reactivity.esm-bundler.js:352:17)
    at Object.set2 [as set] (reactivity.esm-bundler.js:509:17)
    at Proxy.handleUpdate (SocialAuth.vue:72:44)
    at onUpdate:modelValue (SocialAuth.vue:122:62)
```

Tried searching more about this specific error online, but couldn't find any possible solution. This was just a warning and the program wasn't breaking but I found a workaround which is not throwing that warning and state is updating properly. For some reason, properties of object in array are still holding reference to store I guess, I'm not sure. Will need to research and debug more on this. 

This workaround works well, updates the store and doesn't throw any warnings or error. I removed the warning to avoid introducing any possible hidden bugs in the code. Let me know if you guys have any better workaround for this

Reply:
Sketch from our discussion:

```js
function handleUpdate(
  verifier: SocialAuthVerifier,
  inputType: SocialAuthField,
  value: string
) {
  const existingAuth = authDetails.value.find(...)
  if (existingAuthIndex) {
    const existingAuth = authDetails.value[existingAuthIndex];
    const areOtherInputTypesEmpty: SocialAuthField[] = socialAuthTypes
      .filter((el) => el !== inputType)
      .every((type) => !existingAuth[type]);
    if (!value && areOtherInputTypesEmpty) {
      store.commit("clearSocialAuth", ...)
    } else {
      store.commit("addSocialAuth", ...)
    }
  } else {
    store.commit("updateSocialAuth", ...);
  }
}
```